client/src/featureform/register.py

```diff
@@ -1413,6 +1430,51 @@ def register_s3(self,
         self.__resources.append(provider)
         return FileStoreProvider(self, provider, s3_config, s3_config.type())
 
+    def register_gcs(self,
+                    name: str,
+                    credentials: GCPCredentials,
+                    bucket_name: str,
+                    bucket_path: str = "",
+                    description: str = "",
+                    team: str = "", ):
+        """Register a GCS store provider.
+
+        This has the functionality of an offline store and can be used as a parameter
+        to a k8s or spark provider
+
+        **Examples**:
+        ```
+        gcs = ff.register_gcs(
+            name="gcs-quickstart",
+            credentials=gcp_creds,
+            bucket_name="bucket_name",
+            bucket_path="featureform/path/",
+            description="An gcs store provider to store offline"
+        )
+        ```
+        Args:
+            name (str): Name of GCS store to be registered
+            credentials (GCPCredentials): GCP credentials to access the bucket
+            bucket_name (str): the bucket name
+            bucket_path (str): custom path to be used by featureform
+            description (str): Description of Redis provider to be registered
+            team (str): team with permission to this storage layer
```

Comment:
Should change this for all the filestores. The team is just the name of the team registering the filestore